### PR TITLE
TST: spatial.directed_hausdorff: Parametrized test_random_state_None_int

### PR DESCRIPTION
**Pull request by matthewborish**
#### Reference issue

There is no reference issue. This update was a recommendation at scipy conference sprint. 

#### What does this implement/fix?

Parametrized test_random_state_None_int function

### CODE REVIEW

**Review by tupui — APPROVED** [review]
LGTM, thank you @matthewborish! I will wait for a bit of CI to return back before merging.

**Review by tylerjereddy — APPROVED** [review]
LGTM as well, and I suggested the change as a reasonable "onboarding issue" at the sprint.

Note that we have at times in the past had debates about the performance drawbacks of making many more tests from parametrizing vs. old-school loops, even with the reporting advantages of param tests. I think we can consider that out of scope here; I believe we've mostly favored parametrization of late.

**Pull request merged**
